index.ts

```diff
@@ -55,6 +57,21 @@ function useLocalStorage<T>(
     }
   }, [storedValue]);
 
+  useEffect(() => {
+    const handleStorageChange = (e: StorageEvent) => {
+      if (!syncData || e.key !== key) return;
+
```

Comment:
Do we need to do an additional check to make sure the storage event is indeed for localstorage? Something like `e.storageArea == localStorage`?

Reply:
Yeah that is a great idea.